AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/InitializeMutation.dfy

```diff
@@ -576,24 +578,24 @@ module {:options "/functionSyntax:4" } InternalInitializeMutation {
         CreateTime := timestamp,
         CiphertextBlob := [0] // [0] is a temporary place holder, but we should fix this by creating an internal type
       );
-      var SignedMutationIndex :- SystemKeyHandler.SignIndex(newIndex, systemKey);
+      var signedMutationIndex :- SystemKeyHandler.SignIndex(newIndex, systemKey);
+      assert CommitmentAndIndex.ValidIndex?(signedMutationIndex);
```

Comment:
Why is assert necessary here? Can't we do `:- Need(CommitmentAndIndex.ValidIndex?(signedMutationIndex), Error)`

Reply:
Assert is nice because it let's me check that I do not need a runtime check.
A need will get compiled into the runtime artifact;
while runtime checks are often necessary,
I am using formal verification to prove that such a runtime check is redundant.